### PR TITLE
#979 Adding docs for updating template / cm for deployment on different infrastructure

### PR DESCRIPTION
**Pull request by ibuziuk**
### What does this PR do?
Adding docs for updating template / cm for deployment on different infrastructure

### What issues does this PR fix or reference?
#979 

image

### How have you tested this PR?
N/A

### CODE REVIEW

**Comment by ibuziuk** [comment]
[test]

File: dev-guide.adoc [hunk 1]
```diff
@@ -313,6 +313,56 @@ docker push ${DOCKER_IMAGE}:${DOCKER_TAG} && \
...
+
+[NOTE]
+====
+The same https://github.com/redhat-developer/rh-che/blob/master/openshift/rh-che.app.yaml[rh-che.app.yaml] openshift template is used for `rh-che` deployment on both `dsaas-stg / dsaas` (prod-preview / prod). One should be really carefull with the template update, since it would affect production regardless of the https://github.com/openshiftio/saas-openshiftio/blob/master/dsaas-services/rh-che6.yaml#L2[hash] commit defined in the https://github.com/openshiftio/saas-openshiftio/blob/master/dsaas-services/rh-che6.yaml[saas-openshiftio] (each time during the rollout update, the template is used directly from the master branch of the original repository). However, `ConfigMap` https://github.com/redhat-developer/rh-che/blob/master/openshift/rh-che.config.yaml[rh-che.config.yaml] is `NOT` used for deployment on `dsaas-stg / dsaas` (prod-preview / prod). In order to update the `ConfigMap` one needs to create a dedicated HK (housekeping) issue and ask someone from SD (Service Delivery) team to apply new `ConfigMap` values on `dsaas-stg / dsaas`. Example of the HK https://gitlab.cee.redhat.com/dtsd/housekeeping/issues/2527[issue].
```
Comment (amisevsk) [inline dev-guide.adoc]: Type: houseke**e**ping

Comment (ibuziuk) [reply dev-guide.adoc]: good catch. Fixed

**Comment by ibuziuk** [comment]
@ScrewTSW @rhopp looks like there are problems with functional tests

> [INFO] 
> [INFO] Results:
> [INFO] 
> [ERROR] Failures: 
> [ERROR]   BuildAndRunProjectTest.checkWorkspace » Provision Unable to provision, see the...
> [ERROR]   WorkspaceTest.testCorrectWayOfStarting:46 » Timeout Expected condition failed:...
> [INFO] 
> [ERROR] Tests run: 4, Failures: 2, Errors: 0, Skipped: 1

**Pull request merged**